app/controllers/summoners_controller.rb

```diff
@@ -12,18 +12,22 @@ def create
     name = summoner_params[:name]
     summoner_info = Riot.summoner_name(name)[name]
     params = {"riot_id" => summoner_info['id'], "name" => summoner_info['name']}
-    @summoner = Summoner.new(params)
-    if @summoner.save
-      flash[:notice] = 'summoner added'
-      teams_array = Riot.team(params["riot_id"])[@summoner.riot_id.to_s]
-      teams_array.each do |team|
-        Team.make(team, @summoner)
-      end
-      redirect_to summoner_path(@summoner)
+    if Summoner.exists?(riot_id: summoner_info['id'])
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.